src/components/RegisterForm.vue

```diff
@@ -0,0 +1,190 @@
+    <div v-else-if="fnameValidation" > <input v-model="fname" class="Valid"></div>
+    <div v-else><p>*name is not valid*</p><input v-model="fname" class="NoValid"></div>
+
+    <h2>Mail:</h2>
+    <div v-if="emptyMail">*please enter your email*<br><br><input v-model="mail"></div>
+    <div v-else-if="emailValidation" > <input v-model="mail" class="Valid"></div>
+    <div v-else><p>*email is not valid*</p><input v-model="mail" class="NoValid"></div>
+
+    <h2>Phone Number:</h2>
+    <div v-if="emptyPhoneNumber">*please enter phone number*<br><br><input v-model="phonenum"></div>
+    <div v-else-if="phonenumValidation"> <input v-model="phonenum" class="Valid"></div>
+    <div v-else><p>*number is not valid*</p><input v-model="phonenum" class="NoValid"></div>
+    
+    <h2>Gender:</h2>
+    <div v-if="genderValidation">
+    <select v-model="gender" class="Valid">
+      <option disabled value=""></option>
+      <option value="Male">Male</option>
+      <option value="Female">Female</option>
+      <option value="Both">Both</option>
+    </select></div>
+    <div v-else>*gender has to be selected*<br><br>
+    <select v-model="gender">
+      <option disabled value=""></option>
+      <option value="Male">Male</option>
+      <option value="Female">Female</option>
+      <option value="Both">Both</option>
+    </select></div>
+
+    <h2>Your Avatar:</h2>
+    <label v-if="imgNoValidation">*please select an Avatar*<br><br></label>
+    <label v-for="img in imgs" :key="img.src">
+      <input type="radio" v-model="image" :value="img.value">
+      <img :src="img.src" alt="avatar" /></label><br><br>
+
+    <button @click.prevent="submit"> Submit</button> <br><br> 
+    <hr size="3" noshade>
+  
+  </div>
+</template>
+
+<script>
+export default {
+  data: function() {
+    return {
+      fname: '',
+      mail: '',
+      phonenum: '',
+      gender: '',
+      image: '',
+      msg: 'Register Here',
+      imgs:[
+        {src: require("../assets/icons/dog.png") , value:"dog"},
+        {src: require("../assets/icons/fish.png"), value:"fish"},
+        {src: require("../assets/icons/owl.png"), value:"owl"}
+      ]
+    };
+  },
+  methods: {
+    submit: function(){
+      if(this.emailValidation && this.phonenumValidation&&
+         this.fnameValidation && this.genderValidation&&
+         !this.imgNoValidation)
+      {
+        alert('submitted');
+        console.log(this.jsonOutput);
+        this.cleanForm()
+      }
+      else{
+        alert('oops');
+      }
+    },
+    cleanForm: function(){
+      this.fname = '';
+      this.mail = '';
+      this.phonenum = '';
+      this.gender = '';
+      this.image = '';
+    }
+  },
+  computed: {
+    // fnameCaps: function() {
+    //   return this.fname.toUpperCase();
+    fnameValidation: function(){
+      if(this.fname.match(/^([a-zA-Z]{1,} )([a-zA-Z]{1,})$/))
+        return true;
+      else
+        return false;
+    },
+    emailValidation: function () {
```

Comment:
`return this.mail.test(/^\w+@[a-zA-Z_]+?\.[a-zA-Z]{2,3}$/);`
and that applies to all of your validation functions - no need to write 4 lines if you can do it in 1 line without losing any readability